core/src/main/java/org/geysermc/geyser/command/defaults/ConnectionTestCommand.java

```diff
@@ -55,31 +64,69 @@ public void execute(@Nullable GeyserSession session, GeyserCommandSource sender,
+        String[] fullAddress = args[0].replace("<", "").replace(">", "").split(":", 2);
+
         // Still allow people to not supply a port and fallback to 19132
-        String[] fullAddress = args[0].split(":", 2);
         int port;
         if (fullAddress.length == 2) {
-            port = Integer.parseInt(fullAddress[1]);
+            try {
+                port = Integer.parseInt(fullAddress[1]);
+            } catch (NumberFormatException e) {
+                // can occur if e.g. "/geyser connectiontest <ip>:<port> is ran
+                sender.sendMessage("Not a valid port! Specify a valid numeric port.");
+                return;
+            }
         } else {
             port = 19132;
         }
+        String ip = fullAddress[0];
+
+        // Issue: people commonly checking placeholders
+        if (ip.equals("ip")) {
+            sender.sendMessage(ip + " is not a valid IP, and instead a placeholder. Please specify the IP to check.");
+            return;
+        }
+
+        // Issue: checking 0.0.0.0 won't work
+        if (ip.equals("0.0.0.0")) {
+            sender.sendMessage("Please specify the IP that you would connect with. 0.0.0.0 in the config tells Geyser to the listen on the server's IPv4.");
+            return;
+        }
+
+        // Issue: people testing local ip
+        if (ip.equals("localhost") || ip.startsWith("127.") || ip.startsWith("10.") || ip.startsWith("192.168.")) {
+            sender.sendMessage("This tool checks if connections from other networks are possible, so you cannot check a local IP.");
+            return;
+        }
 
         // Issue: do the ports not line up?
         if (port != geyser.getConfig().getBedrock().port()) {
-            sender.sendMessage("The port you supplied (" + port + ") does not match the port supplied in Geyser's configuration ("
-                    + geyser.getConfig().getBedrock().port() + "). You can change it under `bedrock` `port`.");
+            if (fullAddress.length == 2) {
+                sender.sendMessage("The port you are testing with (" + port + ") is not the same as you set in your Geyser configuration ("
+                    + geyser.getConfig().getBedrock().port() + ")");
+                sender.sendMessage("Re-run the command with the port in the config, or change the `bedrock` `port` in the config.");
+                if (geyser.getConfig().getBedrock().isCloneRemotePort()) {
+                    sender.sendMessage("You have `clone-remote-port` enabled. This option ignores the `bedrock` `port` in the config, and uses the Java server port instead.");
+                }
+            } else {
+                sender.sendMessage("You did not specify the port to check (add it with \":<port>\"), " +
+                        "and the default port 19132 does not match the port in your Geyser configuration ("
+                        + geyser.getConfig().getBedrock().port() + ")! ");
```

Comment:
is there a space missing before the "and"?

snd remove the space after the "!"

Reply:
fixed in https://github.com/GeyserMC/Geyser/commit/4b50a606041d11b71c1a5665904d13f8c4fbe295